common/include/pcl/point_cloud.h

```diff
@@ -506,22 +544,60 @@ namespace pcl
         height = 1;
       }
 
+      /**
+       * \brief Replaces the points with `new_width * new_height` copies of `value`
+       * \param[in] new_width new width of the point cloud
+       * \param[in] new_height new height of the point cloud
+       */
+      void
+      assign(index_t new_width, index_t new_height, const PointT& value)
+      {
+        points.assign(new_width * new_height, value);
+        width = new_width;
+        height = new_height;
+      }
+
       /**
        * \brief Replaces the points with copies of those in the range `[first, last)`
        * \details The behavior is undefined if either argument is an iterator into
        * `*this`
        * \note This breaks the organized structure of the cloud by setting the height to
        * 1!
        */
-      template <class InputIt>
+      template <class InputIterator>
       void
-      assign(InputIt first, InputIt last)
+      assign(InputIterator first, InputIterator last)
       {
         points.assign(std::move(first), std::move(last));
         width = static_cast<std::uint32_t>(size());
         height = 1;
       }
 
+      /**
+       * \brief Replaces the points with copies of those in the range `[first, last)`
+       * \details The behavior is undefined if either argument is an iterator into
+       * `*this`
+       * \note This calculates the height based on size and width provided. This means
+       * the assignment happens even if the size is not perfectly divisible by width
+       * \param[in] new_width new width of the point cloud
+       */
+      template <class InputIterator>
+      void
+      assign(InputIterator first, InputIterator last, index_t new_width)
+      {
+        points.assign(std::move(first), std::move(last));
+        width = new_width;
+        height = size() / width;
+        if (width * height != size()) {
+          PCL_WARN("Mismatch in assignment. Requested width (%zu) doesn't divide "
+                   "provided size (%zu) cleanly. Setting height to 1",
+                   static_cast<std::size>(width),
```

Comment:
```suggestion
                   static_cast<std::uint32_t>(width),
```

As in line 572?

Reply:
or index_t ?

Reply:
Oops, size_t. This is for printing and needs to be the largest possible since the format is kinda fixed. There's a macro as well, but that's very weird to use
```suggestion
                   static_cast<std::size_t>(width),
```